File: Python/cards/validation.py
def validCard(card):
    card = [int(x) for x in str(card)]
    card = card[::-1]
    for i in range(1, len(card), 2):
        card[i] *= 2
        if card[i] > 9:
            card[i] -= 9
    return sum(card) % 10 == 0

def validModN(card, n):
    card = [int(x) for x in str(card)]
    card = card[::-1]
    for i in range(1, len(card), 2):
        card[i] *= 2
        if card[i] > 9:
            card[i] -= 9
    return sum(card) % n == 0 

def main():
    card = input('Enter a card number: ')
    if validCard(card):
        print('Valid card')
    else:
        print('Invalid card')

main()

File: Golang/cards/validation.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var cardNumber string
	fmt.Print("Enter a card number: ")
	fmt.Scanln(&cardNumber)
	fmt.Println("Valid: ", validate(cardNumber))
}

func validate(cardNumber string) bool {
	var sum int
	for i := 0; i < len(cardNumber); i++ {
		digit, _ := strconv.Atoi(string(cardNumber[i]))
		if i%2 == 0 {
			digit *= 2
			if digit > 9 {
				digit -= 9
			}
		}
		sum += digit
	}
	return sum % 10 == 0
}
